ShoppingM8-be/src/main/java/edu/pw/shoppingm8/list/api/ListController.java

```diff
@@ -0,0 +1,104 @@
+package edu.pw.shoppingm8.list.api;
+
+import java.net.URI;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import edu.pw.shoppingm8.authentication.AuthenticationService;
+import edu.pw.shoppingm8.list.List;
+import edu.pw.shoppingm8.list.ListService;
+import edu.pw.shoppingm8.list.api.dto.ListDto;
+import edu.pw.shoppingm8.user.User;
+import io.swagger.annotations.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+
+@RestController
+@RequestMapping("list")
+@RequiredArgsConstructor
+@Slf4j
+public class ListController {
+    private final ListService listService;
+    private final AuthenticationService authenticationService;
+
+    @ApiOperation(value = "Create list", nickname = "create list", notes = "",
+            authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "If list was created", response = ListDto.class),
+        @ApiResponse(code = 400, message = "If request has list id or owner id set")})
+    @PostMapping(value="/new")
+    ResponseEntity<ListDto> createList(@RequestBody ListDto listDto) {
+        if (listDto.getId() != null || listDto.getOwnerId() != null)
+            return ResponseEntity.badRequest().build();
+
+        List created = listService.create(listDto);
+        return ResponseEntity.created(URI.create("/list/" + created.getId()))
+            .body(ListDto.of(created));
+    }
+
+    @ApiOperation(value = "Get list", nickname = "get list", notes = "",
+        authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "If list was updated", response = ListDto.class),
+        @ApiResponse(code = 403, message = "If user has no access to this list"),
+        @ApiResponse(code = 404, message = "If list does not exist"),
+    })
+    @GetMapping(value="/{id}")
+    ResponseEntity<ListDto> getList(@PathVariable Long id) {
+        User authenticatedUser = authenticationService.getAuthenticatedUser();
+
+        List list = listService.getList(id);
+        if (!authenticatedUser.getId().equals(list.getOwner().getId()))
```

Comment:
This check is part of the business logic. Please, move it to the service class. Also, I would like to use exception driven approach of handling situations like these. Please, consider creating a dedicated exception and use it in DefaultExceptionHandler which is already provided

Reply:
OK, I will change it.